cmd/argo/commands/cp.go

```diff
@@ -0,0 +1,143 @@
+package commands
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient"
+	workflowpkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflow"
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+type copyArtifactOpts struct {
+	namespace    string // --namespace
+	nodeId       string // --node-id
+	templateName string // --template-name
+	artifactName string // --artifact-name
+	path         string // --path
+}
+
+func NewCpCommand() *cobra.Command {
+	var cpArtifactOpts copyArtifactOpts
```

Comment:
minor - but I think you could just inline all you options rather than using a struct - code would be simpler

Reply:
replaced cpArtifactOpts with direct variables